Libraries/Animated/src/AnimatedImplementation.js

```diff
@@ -753,9 +755,11 @@ class AnimatedValue extends AnimatedWithChildren {
     this._tracking = tracking;
   }
 
-  _updateValue(value: number): void {
+  _updateValue(value: number, flush: bool): void {
     this._value = value;
-    _flush(this);
+    if (flush) {
```

Comment:
It could, but than I'd have to add another check for `isNative` in `onUpdate` handler [here](https://github.com/facebook/react-native/blob/cd1173881982d21b7048cfd65b008238a23a9a95/Libraries/Animated/src/AnimatedImplementation.js#L733) to make sure that this code path does not execute for non-native nodes.

So I actually prefer to only check that in one place [where the check it currently is](https://github.com/facebook/react-native/blob/cd1173881982d21b7048cfd65b008238a23a9a95/Libraries/Animated/src/AnimatedImplementation.js#L1381) and in the remaining places be explicit about whether we want a particular code path to execute for non-native vs native nodes. E.g., it is more likely that someone will add a call to `_updateValue` in a code path that should execute for non-native nodes only and than native implementation won't show any warnings (because we won't call `flush`). Having that variable there will make that person think of what value should be passed (if code should execute for non-native nodes only just pass `true` and if native node will be accident execute that code path it will throw)

So that's just my preference :) If you feel strongly about changing it the way you described I'll do so